### DISCUSSION TITLE
Detecting changes in Local similator variables

### DISCUSSION THREAD

**Discussion by EcstaticAviator1** [mpaperno/MSFSTouchPortalPlugin#43] [Q&A] [answered]
Hi there,

Thought I'd get in touch about the possibility of adding a new Event similar to what is already an Action.

I use the Action "Set Local Airplane Variable" quite a lot, but I also really need an Event "When local airplane variable". So that I can make the button change icon depending on what the local airplane variable is.

Unless you can point me in the direction of another easy way of doing this!

Thanks heaps

**Comment by mpaperno** [comment, marked as answer]
Hi,

To Touch Portal, all the sim variable types are just sent as TP States.  So you'd use "when plugin state changes..." TP event to monitor changes.

Or maybe I don't understand what you mean?

How to request new variables as TP States (including Local ones) is described here:  https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/Using-Custom-States-and-Simulator-Variables

Keep in mind I cannot make new events like "when plugin state changes" because this is a TP feature, not something a plugin can add.

Cheers,
-Max

**Reply by EcstaticAviator1** [reply]
Ahh okay I understand.

So I need to make a configuration ini file to add the needed local vars into the list?

once I have created a ini file, where does it need to go in the plug-ins directory?

There’s no video tutorials on this process?

thanks

**Reply by mpaperno** [reply]
Yes the ini file version is the best and most flexible way, as long as you get it in there correctly.  It may help to create a few variable requests using the provided actions (perhaps using the [editor example page](https://github.com/mpaperno/MSFSTouchPortalPlugin/wiki/Pages-Buttons-and-Graphics#page-template-for-editing-custom-states-and-simulator-variables)) and then save those to file (with the provided plugin action) and go from there.  This will also by default create the file in the right location.

Your custom ini file could go anywhere but by default it is expected to be in the plugin's config folder, "is in the current users's AppData\Roaming\MSFSTouchPortalPlugin folder".  This is in the 2nd paragraph of the documentation I linked to.

I do not make video tutorials and if anyone else did, they haven't told me.

-Max
